### PR TITLE
added a check for all sensors being off before going into ready in teleopinit

### LINKED ISSUE #604: Update Robot.Cpp Notemanager initial state
In telop Init of Robot.cpp we set the current state to Ready. (Shown Below) We need to make the state Ready only if no sensor are Detected.

   auto config = RobotConfigMgr::GetInstance()->GetCurrentConfig();
    if (config != nullptr)
    {
        auto noteMgr = config->GetMechanism(MechanismTypes::MECHANISM_TYPE::NOTE_MANAGER);
        if (noteMgr != nullptr)
        {
            noteMgr->SetCurrentState(noteManagerGen::STATE_NAMES::STATE_READY, true);
        }
    }

### PR DESCRIPTION
**Pull request by JamesLimothy**
closes #604

### CODE REVIEW

**Comment by Tan560** [comment]
@witcpalek  This all looks good to me, but I don't know if I am missing something/we should be doing this a different way. Please check and approve if you agree.

Thanks

File: src/main/cpp/Robot.cpp [hunk 1]
```diff
@@ -122,7 +122,14 @@ void Robot::TeleopInit()
     if (config != nullptr)
     {
         auto noteMgr = config->GetMechanism(MechanismTypes::MECHANISM_TYPE::NOTE_MANAGER);
-        if (noteMgr != nullptr)
+        bool allSensorsOff = ((m_noteManagerGen->getfeederSensor()->Get() == false) &&
+                              (m_noteManagerGen->getlauncherSensor()->Get() == false) &&
```
Comment (witcpalek) [inline src/main/cpp/Robot.cpp:126]: inconsistent variable names; m_noteManagerGet has not been initialized yet.  There is no guarantee that getxxxsensor will not return a nullptr, so it needs to be checked before calling the get method.  

Then  boolean compares don't need == false, just user ! before it

I don't think you need a classwide variable either

File: src/main/cpp/Robot.h [hunk 2]
```diff
@@ -73,5 +74,6 @@ class Robot : public frc::TimedRobot
     AutonPreviewer *m_previewer;
     RobotState *m_robotState;
     SomeMech *m_someMech;
+    noteManagerGen *m_noteManagerGen;
```
Comment (witcpalek) [inline src/main/cpp/Robot.h:77]: Don't need this file changed, just use a local variable in the one method that needs the notemanager

**Review by witcpalek — CHANGES_REQUESTED** [review]

**Comment by Tan560** [comment]
@witcpalek  Please pull this down too and we can test this tomorrow, I have merged it with main

**Review by Tan560 — APPROVED** [review]

**Comment by Tan560** [comment]
Tested on 9999 and works

**Review by witcpalek — APPROVED** [review]

**Pull request merged**